File: Week4/structs1.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>

// include our own file in the header
#include "structs.h"

// define a constant with a value of 3
#define STUDENTS 3

int main(void)
{
    // value type student; array students; size STUDENTS or 3
    student students[STUDENTS];
    
    for (int i = 0; i < STUDENTS; i++) 
    {
        printf("name: ");
        students[i].name = get_string();
        
        printf("dorm: ");
        students[i].dorm = get_string();
    }
    
    // get a variable with pointer to a file
    // open it in w-write mode
    FILE *file = fopen("students.csv", "w");
    if (file != NULL)
    {
        for (int i = 0; i < STUDENTS; i++) 
        {
            fprintf(file, "%s,%s\n", students[i].name, students[i].dorm);
        }
        fclose(file);
    }
}

File: pset6/structs1.py
import cs50
import csv
from student import Student

# mutable list, can be changed later, unlike lists in C
students = []

for i in range(3):
    print("name: ", end="")
    name = cs50.get_string()
    
    print("dorm: ", end="")
    dorm = cs50.get_string()
    
    # append is a method of a list
    students.append(Student(name, dorm))
    
file = open("students.csv", "w")
writer = csv.writer(file)
for student in students:
    writer.writerow((student.name, student.dorm))
file.close()    